api/types.proto

```diff
@@ -713,3 +713,33 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+	NODE = 1 [(gogoproto.enumvalue_customname) = "NodeSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretDataID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_data_id = 2 [(gogoproto.customname) = "SecretDataID"];
```

Comment:
I suggested using opaque IDs instead of the digest because the digest leaks information about the secret itself.

For example if the digest is `sha256:ef92b778bafe771e89245b89ecbc08a44a4e166c06659911881f383d4473e94f`, it's rather straightforward to find out that the secret is `password123`.